Library/Homebrew/os.rb

```diff
@@ -1,6 +1,6 @@
 module OS
   def self.mac?
-    /darwin/i === RUBY_PLATFORM
+    /darwin/i === RUBY_PLATFORM || ENV["HOMEBREW_EMULATE_MAC"]
```

Comment:
Yes, it is still not perfect. But it's currently workable for most of meta data. I know some users invoking brew info command inside docker to get bottle URL list for unofficial mirror.